packages/manager/src/features/Firewalls/FirewallDetail/Rules/FirewallRuleTable.tsx

```diff
@@ -278,88 +290,101 @@ const FirewallRuleTableRow = React.memo((props: FirewallRuleTableRowProps) => {
 
   const actionMenuProps = {
     disabled: status === 'PENDING_DELETION' || disabled,
-    idx: id,
+    idx: index,
     triggerCloneFirewallRule,
     triggerDeleteFirewallRule,
     triggerOpenRuleDrawerForEditing,
   };
 
+  const {
+    active,
+    attributes,
+    isDragging,
+    listeners,
+    setNodeRef,
+    transform,
+    transition,
+  } = useSortable({ id });
+
+  const isActive = Boolean(active);
+
+  // dnd-kit styles
+  const rowStyles = {
+    '& td': {
+      // Highly recommend to set the `touch-action: none` for all the draggable elements-
+      // in order to prevent scrolling on mobile devices.
+      // refer to https://docs.dndkit.com/api-documentation/sensors/pointer#touch-action
+      touchAction: 'none',
+    },
+    cursor: isDragging ? 'grabbing' : 'grab',
+    position: 'relative',
+    transform: CSS.Transform.toString(transform),
```

Comment:
There is a weird scaling effect when rows have different heights:

https://github.com/user-attachments/assets/cf8d3d3d-4dc7-4ada-98ec-5e393fbfe545

The following solves it, taken from [here](https://github.com/clauderic/dnd-kit/issues/117):

```suggestion
    transform: CSS.Translate.toString(transform),
```

Reply:
good catch 👍